cli/wallet/wallet.go

```diff
@@ -542,25 +555,50 @@ func importMultisig(ctx *cli.Context) error {
 		}
 	}
 
-	var label *string
 	if ctx.IsSet("name") {
 		l := ctx.String("name")
 		label = &l
 	}
-	acc, err := newAccountFromWIF(ctx.App.Writer, ctx.String("wif"), wall.Scrypt, label, pass)
-	if err != nil {
-		return cli.NewExitError(err, 1)
-	}
 
-	if err := acc.ConvertMultisig(m, pubs); err != nil {
-		return cli.NewExitError(err, 1)
+	var accounts = make([]*wallet.Account, 0, 1)
+	for _, pub := range pubs {
+		for _, wallAcc := range wall.Accounts {
+			if wallAcc.ScriptHash().Equals(pub.GetScriptHash()) {
+				accounts = append(accounts, wallAcc)
+				accEncPub = pub
+			}
+		}
 	}
```

Comment:
Remove `accounts`, you don't use them for useful things. Set `acc` once you find the account, and after this if the second match occurs, then set `acc` to nil and immediately break the loops. After that is `acc` is not nil, then it means that proper single account was found. If `acc` is nil, then fallback to WIF handling.